### PR TITLE
Monica/fix ui spacing

### PR DESCRIPTION
**Pull request by myo3**
**Spacing issues addressed:**

- route line resizes based on if stop label is single (20 px) or multiline (28 px)
- when there is no live data on card, spacing from travel time to route diagram has been fixed so has proper spacing
- spacing from the bottom of the route diagram to the bottom of cell has been fixed so it is the same as space of travel time label from top of cell
- spacing at top of tableview fixed to reflect pixel spacing on Zeplin (12 px)

All thanks to autolayout anchor constraints and stackviews!

**Spacing issues to address in next UI haul:**

- space from From search bar (the one that says "Current Location") to the top of the navigation bar is wrong

### CODE REVIEW

File: TCAT/Cells/RouteTableViewCell.swift [hunk 1]
```diff
@@ -31,121 +31,236 @@ class RouteTableViewCell: UITableViewCell {
...
+    var departureStackView: UIStackView
+    var departureTimeLabel: UILabel
+    var arrowImageView: UIImageView
     
-    let arrowImageViewRightSpaceFromSuperview: CGFloat = 12.0
-    let departureLabelSpaceFromArrowImageView: CGFloat = 8.0
+    var liveStackView: UIStackView
+    var liveLabel: UILabel
+    var liveIndicatorView: LiveIndicator
+    var stretchyFillerView: UIView
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift:44]: streeeeetch

File: TCAT/Cells/RouteTableViewCell.swift [hunk 2]
```diff
@@ -31,121 +31,236 @@ class RouteTableViewCell: UITableViewCell {
...
+        ])
+        
+        NSLayoutConstraint.activate([
+            arrowImageView.heightAnchor.constraint(equalToConstant: arrowImageViewHeight),
+            arrowImageView.widthAnchor.constraint(equalToConstant: arrowImageViewWidth),
+            arrowImageView.centerYAnchor.constraint(equalTo: departureTimeLabel.centerYAnchor)
+        ])
+    }
+    
+    private func setTranslatesAutoresizingMaskIntoConstraints() {
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift:216]: Just a thought, what if you made an array with all of these UIViews, and did a for loop?

    let viewArray = [ 
        ... 
    ]

    viewArray.forEach { $0.translatesAutoresizingMaskIntoConstraints = false }

File: TCAT/Cells/RouteTableViewCell.swift [hunk 3]
```diff
@@ -31,121 +31,236 @@ class RouteTableViewCell: UITableViewCell {
...
+    
+        verticalStackView.translatesAutoresizingMaskIntoConstraints = false
+        topBorder.translatesAutoresizingMaskIntoConstraints = false
+        routeDiagram.translatesAutoresizingMaskIntoConstraints = false
+        bottomBorder.translatesAutoresizingMaskIntoConstraints = false
+        cellSeparator.translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    /// For debugging constraint errors
+    private func setDebugIdentifiers() {
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift]: nice!

File: TCAT/Model/BusStop.swift [hunk 4]
```diff
@@ -24,7 +24,7 @@ class BusStop: Place, CoordinateAcceptor {
     }
     
     override func isEqual(_ object: Any?) -> Bool {
-        if (!super.isEqual(object)){
```
Comment (mattbarker016) [inline TCAT/Model/BusStop.swift:27]: nice

File: TCAT/Views/BusIcon.swift [hunk 5]
```diff
@@ -8,34 +8,73 @@
 import UIKit
 
 enum BusIconType: String {
-    case directionSmall
-    case directionLarge
-    case liveTracking
+    case directionSmall, directionLarge, liveTracking
```
Comment (mattbarker016) [inline TCAT/Views/BusIcon.swift:11]: I like using the enum this way a lot. The one thing is when you list each case on a new line, you can add documentation with the triple-slash comment.

**Review by mattbarker016 — COMMENTED** [review]
A lot of your LiveIndicator code is better structured, so we can go with your version. I'll test on my view and fix any issues.

Some spacing off on constraint activation, but no big deal.

Sweet refactoring!

**Comment by mattbarker016** [comment]
This can be closed, I merged the changes into my newer PR. I'l let you close it so nothing is accidentally lost. Fun git times at 2am wee

File: TCAT/Cells/RouteTableViewCell.swift (hunk 2 above)
Comment (myo3) [reply TCAT/Cells/RouteTableViewCell.swift:216]: True that is shorter

**Comment by myo3** [comment]
@mattbarker016 what spacing is off? i can fix it

**Pull request merged**